### PR TITLE
Evaluating Toxicity in Jigsaw Dataset using Cleanlab Studio

### PR DESCRIPTION
**Pull request by aravindputrevu**

### CODE REVIEW

File: jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb [hunk 1]
```diff
@@ -0,0 +1,7833 @@
...
+        "\n",
+        "However, LLMs learn to generate the knowledge from the data it has seen during the training period. Data play a key role in the quality of LLM responses. LLM responses can be affected by the data which is not checked for safety.\n",
+        "\n",
+        "Imagine a toxic conversation (i.e., a data point) annotated or marked as safe used in LLM training, will have collateral impact on all LLM responses and further fine-tunes of that model.\n",
+        "\n",
+        "Aligning datasets for safer LLM responses is the need of the hour.\n",
+        "\n",
+        "\n",
+        "\n",
+        "In this tutorial, we will be using the `Civil Comments` dataset and loading it directly using the tensorflow datasets. The Civil Comments dataset comprises a diverse collection of user comments extracted from the commenting sections of online platforms. These comments cover a wide range of topics and discussions, reflecting the varied nature of online discourse. The dataset includes annotations for toxicity, identity mentions, and other linguistic attributes, providing valuable insights into the prevalence and nature of harmful behavior in online interactions.\n",
```
[the remark this answers is not in the document]
Comment (jwmueller) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:3466]: also always link to the dataset in the first mention

File: jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb [hunk 2]
```diff
@@ -0,0 +1,7833 @@
...
...ufT2UI6azg0+SGAJmWF8+lsLpVlYSIcT+fYGMtM2eS6rq1RJOEIpx9GpGuEEXro752HEzAAJEfAYOKC2FAC5vtmIh3/lzu+OJmb+bPXf7HZrhNGZkOEuJccOIvB3A596bOy7kQE/eFWgwIYwAgPn57bqsF86CglANg4Yu+8vda7NS7ZARFQTJtnr0zuviHxsDVyKMJ8CxFpNptrVTQQybWcqJdnRk7RYZAcdqNXerMVoiAIrDVuGuXcg90YPGMtAj64/4HHPe5xzWazWCw6T9rDerUjnGFwOVGwxgnaDwFQBPFToebo+361WollU36rHQOwMsA0CMlYM5uf+Pzr3y0CU7liNpEiIq1UTHmNdus9L37tq296FgDM5Mf3vu5Zk9k8GvMHL/zlmPaMZcP2sp17//yVv3XVnouE7Udf/dZbHv+M5Vr5mvMuPW9qR6PVzCSSH/7Ft1y15wK/01JE7lYQwLtf+JpX3Pgz+xYOEtLlu87bMT7V7LSGeghF0FO63KgFxnpabxAFMNAtR0J5IQEDEMtypMqLFXPPHHiKihnKpXAsRbkkxjTlUqCo3w8GkSvMsI2n03/7r1/9xT99JwJWW43AmsCYATPflkUxmUGkVS5aBIh7sQELWRipGNJLrbSntERLQJETFQQAEPG01kq7XRCQhRdKyyzsKa21RoGg4//sY5/4J2/+fbUzr6cKlIrByLW11RF6jKPe8LTBCQ/9GJGuEUYYjlWfWTcK8agErHRQeS1RGhdLhzTHi5m0IHenKrwqcALFNumBr/rR6BuG73dTJ7rDGSIwc9dDNVByZtj2fesGtx8WwNRjH72U8TVbgACiMAhCfyidC7IBAdRQOeT/6O+MWOmbTK5F1N7QxgsCw9q86gKj/3T9e2uvRBEUOD7AAAAgAElEQVSJrM21C8tJCfLAcun9F8PLFteqXvqEdP2E7nRuZhDOD9iqbCZ9oLS8Z+85qVT6lEfljbAV0fW0smwUXkgIsVNBuhDR87xSqeS32vE15eO6kb3X7L0MAQxb17+JiGVOeDFCunD3BeDF3/+ZD6/Uq4/cfQG3m4885wIi6gQ+ACRjiZc/6Wa/0/b9DhE97ZHXAVHgdxqdtrOSvOhxTwsC3zeBc+YjAIs0g/Z5Uzsvmt0DAH7g11oNGvTnh3ASoIgqipvd/FX3jsFhLwAuuUsAWHixwnMlQARNQ
```
Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:3891]: why would you set hyperparameters way up here?

Please move everything that's unrelated to the data itself and Cleanlab much lower down in the notebook to where it's actually needed

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:3478]: this image doesn't help me understand the dataset at all.

You need to: 
- describe this image is if you're going to include it
- include an image of the relevant parts of the dataset (showing text example, human toxicity annotation, etc). We cannot have tutorials/blogs that do not make it clear what the dataset looks like.

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5363]: follow our Studio tutorials in how you format this part (including the link to find key)

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5398]: why did you run this as a tabular project? 

It needs to be a text project, with Cleanlab's Text column chosen as the `text` column in this dataset

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5911]: please never write this. It's: "Cleanlab Studio" always

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:7822]: You don't plan to show any `predict()` calls? It seems not very interesting to end the tutorial with a `fit()` call and no demonstration of how the trained model can actually be used (or accuracy evaluation)

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:7785]: how is this with Cleanset? It looks the same to me as without Cleanset. Even if the code is correct, please make it more clearly discernable what the difference is

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5375]: again lots of writing issues so please proofread everything with chatgpt before asking for me review. It's never: 'the Cleanlab ...", it's just "Cleanlab ..."

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5920]: In all of the results here, it is way too hard to see the `text` column, and the given annotation, which are the primary items of interest. You need to make those more visible. 

Also you didn't discuss the results at all. How is somebody supposed to know what Cleanlab Studio is detecting here?

Comment (jwmueller) [inline jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:7220]: think you mean "data points" not "columns"?

Comment (jwmueller) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:7220]: remove is also an unclear term, 'filter from the dataset' is more clear

File: jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (aravindputrevu) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:3466]: Done, thanks for the catch, inadvertently missed this. added enough context now.

File: jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb (hunk 2 above)
Comment (aravindputrevu) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:3891]: Done!

Comment (aravindputrevu) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:3478]: Ok, I think this one is slightly confusing to as the below section with `pandas` shows how the dataset is structured. I will add few lines to explain the dataset

Comment (aravindputrevu) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5911]: Corrected it.

Comment (aravindputrevu) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:7220]: Rewrote this line

Comment (aravindputrevu) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb:5375]: Sure.

Comment (jwmueller) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb]: you never used your BATCH_SIZE hyperparameter

Screen Shot 2024-04-08 at 9 27 42 PM

**Review by jwmueller — APPROVED** [review]
pushed some edits quickly, but am going to merge this in the interest of time

File: jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb (hunk 2 above)
Comment (jwmueller) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb]: you never addressed this. not sure why. I'm going to just make a guess fix

Comment (jwmueller) [reply jigsaw_ai_safety_keras/Evaluating_Toxicity_Datasets_Large_Language_Models.ipynb]: I pushed fix here:  https://github.com/cleanlab/cleanlab-tools/pull/9

Please LMK if it's wrong
